CMakeLists.txt

```diff
@@ -151,6 +158,11 @@ if (NOT PARTHENON_DISABLE_HDF5)
   target_link_libraries(HDF5_C INTERFACE ${HDF5_C_LIBRARIES})
   target_compile_definitions(HDF5_C INTERFACE ${HDF5_C_DEFINITIONS})
   target_include_directories(HDF5_C INTERFACE ${HDF5_C_INCLUDE_DIRS})
+
+  install(TARGETS HDF5_C EXPORT parthenonTargets
+    INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/parthenon"
+    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
```

Comment:
Experimenting with this, it seems like `INCLUDES`, `LIRBARY`, and `ARCHIVE` are unnecessary, and in fact `    INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/parthenon"` adds `/parthenon` to Parthenon::HDF5_C's `INTERFACE_INCLUDE_DIRECTORIES` property, which recently gave me an error trying to configure an external app. I've gone ahead and replaced this with just `install(TARGETS HDF5_C EXPORT parthenonTargets)` which works fine on my end.

Reply:
@JoshuaSBrown Am I right that you added the HDF5_C install target? Maybe this change is not a good idea

Reply:
I don't honestly remember, but if hdf5 works fine with the parthenon library then it would seem what you are doing is cleaner.